acados/sim/sim_erk_integrator.c

```diff
@@ -748,8 +755,21 @@ int sim_erk(void *config_, sim_in *in, sim_out *out, void *opts_, void *mem_, vo
         // store hessian
         if (opts->sens_hess)
         {
-            for (i = 0; i < nhess; i++)
-                S_hess_out[i] = adj_tmp[nx + nu + i];
+            // former line for tridiagonal export was
+            //            for (i = 0; i < nhess; i++) S_hess_out[i] = adj_tmp[nx + nu + i];
+            int count_upper = 0;
+            for (int j = 0; j < nx + nu; j++) {
+                for (int i = 0; i < nx + nu; i++){
+                    // S_hess_out[i] = adj_tmp[nx + nu + i];
+                    if ( i >= j ) {
```

Comment:
Why not have the inner for loop for(int i=j; i<nx+nu; i++) and remove the if ?

Also, in theory only the lower triangular part of the hessian is needed, so ti should not even be necessary to copy it on the upper triangular part.

Reply:
Add a reference @FreyJo